std/math.d

```diff
@@ -4590,7 +4590,7 @@ public:
     assert(ieeeFlags == f);
 }
 
-@system unittest
+version(D_HardFloat) @system unittest
```

Comment:
The topic is some small changes I had to make for Android/ARM, ie the contents of this PR.  The notion that one can't discuss this tiny commit here because of the surrounding discussion about murmurhash3 is laughable, as most non-negligible Phobos PRs have many such topical threads of discussion.